airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py

```diff
@@ -47,10 +49,12 @@ class SourceGoogleAds(AbstractSource):
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
         try:
             logger.info("Checking the config")
-            GoogleAds(credentials=config["credentials"], customer_id=config["customer_id"])
+            google_api = GoogleAds(credentials=config["credentials"], customer_id=config["customer_id"])
+            account_stream = Accounts(api=google_api)
+            list(account_stream.read_records(sync_mode=SyncMode.full_refresh))
             return True, None
-        except Exception as error:
-            return False, f"Unable to connect to Google Ads API with the provided credentials - {repr(error)}"
+        except GoogleAdsException as error:
```

Comment:
You can remove try/except block, any exception would be catch on AbstractSource check method

Reply:
Yes, but the main goal of the try/except is the correct formulation of the error message.